src/components/utils/camera.ts

```diff
@@ -0,0 +1,50 @@
+import Webcam from 'react-webcam-onfido'
+import { canvasToBlob } from './blob'
+
+import type { SdkMetadata } from '~types/commons'
+import type { HandleCaptureProp } from '~types/routers'
+
+export const screenshot = (
+  webcam: Webcam,
+  callback: (blob: Blob, sdkMetadata: SdkMetadata) => void,
+  mimeType?: string
+): void => {
+  if (!webcam) {
+    console.error('webcam is null')
+    return
+  }
+
+  const canvas = webcam && webcam.getCanvas()
+
+  if (!canvas) {
+    console.error('webcam canvas is null')
+    return
+  }
+
+  const sdkMetadata = getDeviceInfo(webcam.stream)
+  canvasToBlob(canvas, (blob) => callback(blob, sdkMetadata), mimeType)
+}
+
+export const getRecordedVideo = (
+  webcam: Webcam,
+  callback: HandleCaptureProp
+): void => {
+  const blob = webcam.getVideoBlob()
+  const sdkMetadata = getDeviceInfo(webcam.stream)
+  callback({ blob, sdkMetadata })
+}
+
+const getDeviceInfo = (stream: MediaStream): SdkMetadata => {
```

Comment:
`SdkMetadata` could be an empty object, does that need to be specified here or is that already expected in the `SdkMetadata` declaration?

Reply:
Yes, the declaration of [`SdkMetadata` type](https://github.com/onfido/onfido-sdk-ui/blob/improvement/convert-utils-and-tracker-to-ts/src/types/commons.ts#L27-L40) includes optional fields, so an empty object is a legit `SdkMetadata`